soc/xtensa/intel_adsp/ace_v1x/power.c

```diff
@@ -125,14 +223,24 @@ __weak void pm_state_set(enum pm_state state, uint8_t substate_id)
 __weak void pm_state_exit_post_ops(enum pm_state state, uint8_t substate_id)
 {
 	ARG_UNUSED(substate_id);
+	uint32_t cpu = arch_proc_id();
 
 	if (state == PM_STATE_SOFT_OFF) {
-		uint32_t cpu = arch_proc_id();
-
 		/* TODO: move clock gating prevent to imr restore vector when it will be ready. */
 		DFDSPBRCP.bootctl[cpu].bctl |= DFDSPBRCP_BCTL_WAITIPCG;
 		soc_cpus_active[cpu] = true;
 		z_xtensa_cache_flush_inv_all();
+	} else if (state == PM_STATE_SUSPEND_TO_IDLE) {
+		ACE_PWRCTL->wpdsphpxpg |= BIT(cpu);
+		DFDSPBRCP.bootctl[cpu].bctl |=
+			DFDSPBRCP_BCTL_WAITIPCG | DFDSPBRCP_BCTL_WAITIPPG;
+		if (cpu == 0) {
+			DFDSPBRCP.bootctl[cpu].battr &= (~LPSCTL_BATTR_MASK);
+		}
+
+		soc_cpus_active[cpu] = true;
+		z_xtensa_cache_flush_inv_all();
+		z_xt_ints_on(core_desc[cpu].intenable);
```

Comment:
@ceolin this is big commit but I didn't find nice way to split it. Is it ok?

Reply:
Yep, that is totally fine IMHO. You have split the changes into different commits.